### PR TITLE
[improve][broker] Make splitNamespaceBundle and getTopicHashPositions async 

### PR DESCRIPTION
**Pull request by gaozhangmin**
### Motivation

See PIP https://github.com/apache/pulsar/issues/14365 and change tracker https://github.com/apache/pulsar/issues/15043.

Make the below method async:

`getBundlesData`  `splitNamespaceBundle` `getTopicHashPositions` `setPublishRate` `getPublishRate` `setDispatchRate`

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [x] `doc-not-needed`

### CODE REVIEW

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 1]
```diff
@@ -892,15 +899,29 @@ public void splitNamespaceBundle(
...
+                    asyncResponse.resume(Response.noContent().build());
+                })
+                .exceptionally(ex -> {
+                    if (!isRedirectException(ex)) {
+                        log.error("[{}] Failed to split namespace bundle {}/{} due to {}",
+                                clientAppId(), namespaceName, bundleRange, ex.getMessage());
+                    }
+                    Throwable realCause = FutureUtil.unwrapCompletionException(ex);
+                    if (realCause instanceof IllegalArgumentException) {
+                        log.error("[{}] Failed to split namespace bundle {}/{} due to {}",
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:916]: Log repeatedly, you should remove this line.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 2]
```diff
@@ -817,15 +822,28 @@ public void splitNamespaceBundle(
...
+                    asyncResponse.resume(Response.noContent().build());
+                })
+                .exceptionally(ex -> {
+                    if (!isRedirectException(ex)) {
+                        log.error("[{}] Failed to split namespace bundle {}/{} due to {}",
+                                clientAppId(), namespaceName, bundleRange, ex.getMessage());
+                    }
+                    Throwable realCause = FutureUtil.unwrapCompletionException(ex);
+                    if (realCause instanceof IllegalArgumentException) {
+                        log.error("[{}] Failed to split namespace bundle {}/{} due to {}",
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:838]: Log repeatedly, you should remove this line.

**Review by nodece — CHANGES_REQUESTED** [review]

**Review by nodece — APPROVED** [review]

**Pull request merged**